src/app/api/payment/route.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Payment API route
+ *
+ * This route is protected, meaning only authenticated users can access this. Clerk is used to
+ * verify that the user is signed in (see `src/middleware.ts`)
+ */
+
+import { db } from '@/db';
+import { members } from '@/db/schema';
+import { products } from '@/data/products';
+import { env } from '@/env.mjs';
+import { redisClient } from '@/lib/redis';
+import { squareClient } from '@/lib/square';
+import { currentUser } from '@clerk/nextjs';
+import { eq } from 'drizzle-orm';
+import type { NextRequest } from 'next/server';
+import type { CreatePaymentLinkRequest, OrderLineItem } from 'square';
+import { ApiError } from 'square';
+import { z } from 'zod';
+
+// Create a Square payment link
+// See: https://developer.squareup.com/reference/square/checkout-api/create-payment-link
+export async function POST(request: Request) {
+    const req = await request.json();
+    const schema = z.object({
+        product: z.string().min(1),
+        customerId: z.string().min(1),
+        redirectUrl: z.string().url().min(1),
+    });
+
+    // Ensure user is logged in
+    const user = await currentUser();
+    if (!user) {
+        return new Response(null, { status: 401 });
+    }
+
+    const result = schema.safeParse(req);
+    if (!result.success) {
+        return new Response(result.error.message, { status: 400 });
+    }
+
+    let lineItem: OrderLineItem;
+    if (result.data.product === 'membership') {
+        lineItem = products.membership;
```

Comment:
```suggestion
    if (reqBody.data.product !== 'membership') {
```

return first then `lineItem = products.membership;` (so you can save an `else` and modify `let` to `const`)

Reply:
I had originally intended for more products to be added if necessary but I guess since we haven't gotten to that point yet, we can just simplify things and make `lineItem` const.